scripts/dts/gen_defines.py

```diff
@@ -579,6 +583,46 @@ def write_clocks(node):
                 controller.props["clock-frequency"].val)
 
 
+def write_dmas(node):
+    if "dmas" not in node.props:
+        return
+
+    for dma_i, dma in enumerate(node.props["dmas"].val):
+        out_dev_s(node, "DMA_CONTROLLER_{}".format(dma_i), dma.controller.label)
+        out_dev(node, "DMA_CHANNEL_{}".format(dma_i), dma.data["channel"])
+
+        # Client parameter syntax varys with the DMA controller changes.
+        if "st,stm32-dma" in dma.controller.compats:
+            dma.slot = dma.data["slot"]
+            dma.channel_config = dma.data["channel_config"]
+            dma.features = dma.data["features"]
+
+            if dma.channel_config & (0x1 << 9) != 0x0:
+                dma.periph_inc = 0x1
+            else:
+                dma.periph_inc = 0x0
+
+            if dma.channel_config & (0x1 << 10) != 0x0:
+                dma.mem_inc = 0x1
+            else:
+                dma.mem_inc = 0x0
+
+            if dma.channel_config & (0x1 << 15) != 0x0:
+                dma.periph_inc_fixed = 1
+            else:
+                dma.periph_inc_fixed = 0
+
+            dma.priority = (dma.channel_config >> 16) & 0x3
+            dma.fifo_threshold = dma.features & 0x3
+
+            out_dev(node, "DMA_SLOT_{}".format(dma_i), dma.slot)
```

Comment:
Feels strange to add compatible specific code at this point. 
Seems to me we should use binding "#cells" to generate this.
@ulfalizer, would you help sorting this out?

Reply:
I personally also think so, while I didn't find a better way of doing this. There's seems like no examples of having such behaviour.

Reply:
Yeah, would be nicer to use `#cells` here. See the comment I posted above.

Previously, new `phandle-array` properties had to be hardcoded in `edtlib.py`, because I hadn't figured out how to deal with them in a generic way. It was fixed by https://github.com/zephyrproject-rtos/zephyr/pull/19327 though.